### ISSUE TITLE
Don't show "connected to" on connection error

### ISSUE THREAD

**Issue by ekzyis** [getAlby/bitcoin-connect#174]
If there is a connection error, the balance correctly shows a warning sign.

However, it still shows "connected to" and shows the "disconnect" button. This is confusing since it contradicts that there was a connection error.

https://github.com/getAlby/bitcoin-connect/assets/27162016/f20a5fce-b17b-4575-8851-23a801ccc71a

**Comment by rolznz** [comment]
I think this can be fixed by adding validation in the init function of the connector as I mentioned [here](https://github.com/getAlby/bitcoin-connect/issues/175)

> Ideally with LNbits though we would add a connection check to see if the connection is valid (by doing an API call). This can be done in the LNbits connector init function, and if the call fails an error should be thrown which will stop Bitcoin Connect from connecting. https://github.com/getAlby/bitcoin-connect/blob/master/src/connectors/LnbitsConnector.ts#L22

**Comment by ekzyis** [comment]
There is no bounty from me and I am not aware of any bounty

**Comment by 1amKhush** [comment]
On failed connect, should modal stay open with error or close at once?

**Comment by rolznz** [comment]
@1amKhush did you test this already? is it happening on the NWC connector too? if so, would you like to propose a possible solution so it can be reviewed?

I think the user has to somehow see the error message. So the modal should stay open I guess.

**Comment by 1amKhush** [comment]
@rolznz I have tested this in runtime.  Also it does affects the NWC path, not just LNbits/LNC. regarding the error message i agree that the user should be informed. 

How about we add a connector-level `validateConnection()` hook that runs in `store.connect()` immediately after `provider.enable()` and before marking the connection as successful. 

- Each connector can implement this using a lightweight check like `provider.getInfo()`. On failure instaed of calling `disconnect()` we clear the connection state while the modal is open for the user to see i suppose. 

One thing i wanted to ask was how about we make the error message a bit more informative for the user to understand the underlying issue?

**Comment by 1amKhush** [comment]
@rolznz Is it okay if i work on this issue?

**Comment by rolznz** [comment]
I don't think this is the best place. The connect function in the store is run on each page load if a user connected their wallet. Ideally it only tests the connection once when the user initially connects their wallet. Could you have a look and see if you can propose a simple solution before starting coding?

**Comment by 1amKhush** [comment]
> I don't think this is the best place. The connect function in the store is run on each page load if a user connected their wallet. Ideally it only tests the connection once when the user initially connects their wallet. Could you have a look and see if you can propose a simple solution before starting coding?

@rolznz  How about we introduce a connection mode in the store, `mode: 'initial' | 'restore'` (defaulting to `'initial'`), to distinguish between a first-time connection and a session restore. 
- Add a connector-level `validate(provider)` hook with a no-op default; implement it with a lightweight `/api/v1/wallet` call for LNbits and a stricter `getBalance` check for NWC. 
- In `store.connect()`, run validation only in initial mode and set `connected: true` while persisting config only after successful validation. 
- On failure, clear the connection state but keep the modal open instead of calling `disconnect()`. During boot, `use mode: 'restore'` so page reloads skip validation.
